OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs

```diff
@@ -190,14 +190,21 @@ Action InitDisplayPanel(Widget panel)
 
 			var frameLimitTextfield = panel.Get<TextFieldWidget>("FRAME_LIMIT_TEXTFIELD");
 			frameLimitTextfield.Text = ds.MaxFramerate.ToString();
-			frameLimitTextfield.OnLoseFocus = () =>
+			frameLimitTextfield.OnEnterKey = () =>
 			{
 				int fps;
 				Exts.TryParseIntegerInvariant(frameLimitTextfield.Text, out fps);
 				ds.MaxFramerate = fps.Clamp(1, 1000);
 				frameLimitTextfield.Text = ds.MaxFramerate.ToString();
+				frameLimitTextfield.YieldKeyboardFocus();
+				return true;
+			};
+			frameLimitTextfield.OnEscKey = () =>
```

Comment:
In fact if you keep clicking on these text fields alternatingly, you add them multiple times to the stack. And then you can keep pressing ESC and the focus switches from one text field to the other.